### ISSUE TITLE
Pull to refresh and navigation

### ISSUE THREAD

**Issue by markleusink** [unplugged/unplugged-controls#332]
Suppose 10 records show in a list and I use 'pull to refresh' 2 times to show an additional 20 (30 in total). If I then open a record and close it, it returns to the original list containing the first 10 documents. I would expect the controls to remember that it already showed me 30 records, so I can continue were I left of.

Possible solution to this issue would be to update the history object when performing a 'pull to refresh' (adding an extra URL parameter indicating the number of documents that are loaded in total). Additional, the controls could also keep track of your location within the list (maybe using HTML anchors?) so you can return to the exact same position you left off.

**Comment by whitemx** [comment]
The problem is that if we took this further and you'd loaded 300 documents,
the performance of the page load when you go back would be very bad indeed.

On 1 November 2013 08:47, Mark Leusink notifications@github.com wrote:

> Suppose 10 records show in a list and I use 'pull to refresh' 2 times to
> show an additional 20 (30 in total). If I then open a record and close it,
> it returns to the original list containing the first 10 documents. I would
> expect the controls to remember that it already showed me 30 records, so I
> can continue were I left of.
> 
> Possible solution to this issue would be to update the history object when
> performing a 'pull to refresh' (adding an extra URL parameter indicating
> the number of documents that are loaded in total). Additional, the controls
> could also keep track of your location within the list (maybe using HTML
> anchors?) so you can return to the exact same position you left off.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/unplugged/unplugged-controls/issues/332
> .

## 

_Matt White_
Phone - +44 7767 384970
E-Mail - matthew.white@fclonline.com
Personal Blog - mattwhite.me

## Twitter - @mattwhite http://twitter.com/mattwhite

Fynn Consulting Ltd http://www.fclonline.com is a limited company
registered in England & Wales with registered number 4014250

## Registered office address at 24 Ivymount Road, London, SE27 0NB

**Comment by markleusink** [comment]
You're right: I have a test database with 40,000 documents and it takes about 5 seconds to render 300 documents. On the other side: I don't think that users would do a pull to refresh 15 times (assuming 20 recs at a time) to find what they're looking for. A search would be a lot better then.

**Comment by whitemx** [comment]
I think we'll add it to the 3.1 to do list if that's OK. There's a lot of
things we still need to work out for 3.0

On 1 November 2013 15:41, Mark Leusink notifications@github.com wrote:

> You're right: I have a test database with 40,000 documents and it takes
> about 5 seconds to render 300 documents. On the other side: I don't think
> that users would do a pull to refresh 15 times (assuming 20 recs at a time)
> to find what they're looking for. A search would be a lot better then.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/unplugged/unplugged-controls/issues/332#issuecomment-27574847
> .

## 

_Matt White_
Phone - +44 7767 384970
E-Mail - matthew.white@fclonline.com
Personal Blog - mattwhite.me

## Twitter - @mattwhite http://twitter.com/mattwhite

Fynn Consulting Ltd http://www.fclonline.com is a limited company
registered in England & Wales with registered number 4014250

## Registered office address at 24 Ivymount Road, London, SE27 0NB
